### PR TITLE
fix: Update dask version to support pandas 1.x

### PR DESCRIPTION
**Pull request by HaoXuAI**
fix issue: https://github.com/feast-dev/feast/issues/4325

# What this PR does / why we need it:

# Which issue(s) this PR fixes:

# Fixes

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -65,7 +65,7 @@
     "fastapi>=0.68.0",
     "uvicorn[standard]>=0.14.0,<1",
     "gunicorn; platform_system != 'Windows'",
-    "dask[dataframe]>=2024.4.2",
+    "dask[dataframe]>=2024.2.1",
```
Comment (franciscojavierarceo) [inline setup.py:68]: are we downgrading to support pandas 1?

**Comment by tokoko** [comment]
@HaoXuAI Did you try running tests locally with `dask[dataframe]==2024.2.1`? I vaguely remember that we made some code changes in dask offline store when we introduced that lower bound.

File: setup.py (hunk 1 above)
Comment (HaoXuAI) [reply setup.py:68]: not downgrading exactly, just lower the bound of dask to relax the pandas version. I think force pandas to be 2.0+ maybe too tight, will cause conflicts with many other projects.

**Comment by HaoXuAI** [comment]
> @HaoXuAI Did you try running tests locally with `dask[dataframe]==2024.2.1`? I vaguely remember that we made some code changes in dask offline store when we introduced that lower bound.

not sure, which test will fail? looks like it passed the ci. from the pr it doesn't require dask to be specific version: https://github.com/feast-dev/feast/pull/4159/files#diff-c91058a32040621ac54d6ea4194b302a420b7849f6e8561920d23a0e97acf69a

**Review by tokoko — APPROVED** [review]
sure, but CI isn't testing for lower bounds. Anyway, you're right, even if some feature of dask offline store doesn't work, it's not worth it to block users from using pandas v1.

**Pull request merged**